### PR TITLE
[KBV-94] Display questions

### PR DESCRIPTION
**Pull request by merlinc**
### What changed

Add pages to show dynamic questions, loaded from the API

### Issue tracking

- [KBV-XXXX](https://govukverify.atlassian.net/browse/KBV-XXXX)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -3504,6 +3672,19 @@ sinon@11.1.2:
     nise "^5.1.0"
     supports-color "^7.2.0"
 
+sinon@^7.5.0:
```
[the remark this answers is not in the document]
Comment (merlinc) [reply yarn.lock:3675]: I think it's a dependency of Playwright

File: test/utils/mocha-helpers.js [hunk 2]
```diff
@@ -11,3 +15,28 @@ const expect = chai.expect;
 
 global.sinon = sinon;
 global.expect = expect;
+
+global.setupReqResNext = () => {
```
[the remark this answers is not in the document]
Comment (MattBidewell) [reply test/utils/mocha-helpers.js:19]: `SetupMocks`? 

Reasoning - they are mocks of req, res, and next and could potentially be expanded in the future (doubtful)

File: src/app/details/controllers/done.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+const {
+  API: {
+    BASE_URL,
+    PATHS: { AUTHORIZE },
+  },
+} = require("../../../lib/config");
+
+class DoneController extends BaseController {
+  async saveValues(req, res, next) {
```
Comment (MattBidewell) [inline src/app/details/controllers/done.js:12]: We should be doing a callback here with our function logic inside of the callback - 

```
async saveValues(req, res, next) {
   super saveValues(req, res, () => {
       ....
       next();
   }
}
```

File: src/app/details/controllers/person-selector.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
+const BaseController = require("hmpo-form-wizard").Controller;
+
+class PersonSelectorController extends BaseController {
+  locals(req, res, callback) {
+    super.locals(req, res, (err, locals) => {
+      callback(null, locals);
+    });
+  }
+
+  async saveValues(req, res, next) {
```
Comment (MattBidewell) [inline src/app/details/controllers/person-selector.js:10]: Save values is the wrong signature, I believe. ^ same comment as before.

File: src/app/kbv/controllers/load-question.test.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      axios: axiosStub,
+      "../../../lib/config": {
+        API: {
+          BASE_URL: "http://example.com",
+          PATHS: { QUESTION: "/q" },
+        },
+      },
+    });
+
+    req = reqres.req();
```
Comment (MattBidewell) [inline src/app/kbv/controllers/load-question.test.js:33]: From here we can go and call the `global.setupReqResNext()` to save lines of code 🤓

File: src/app/kbv/controllers/question.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+const {
+  API: {
+    BASE_URL,
+    PATHS: { QUESTION, ANSWER },
+  },
+} = require("../../../lib/config");
+
+class QuestionController extends BaseController {
+  configure(req, res, next) {
```
Comment (MattBidewell) [inline src/app/kbv/controllers/question.js:15]: Does configure(..) need to follow the same signature as `saveValues()` and `locals()`

File: src/app/kbv/controllers/question.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        }
+      } catch (e) {
+        debug(e.message);
+      }
+
+      con();
+    });
+  }
+
+  next(req) {
```
Comment (MattBidewell) [inline src/app/kbv/controllers/question.js:114]: ~~Not 100% sure what the intention is here, is this debug code? I don't believe I can see `this.next` being called anywhere~~

Scratch that, I saw further down in steps the use of question.next. Nice!

**Review by MattBidewell — CHANGES_REQUESTED** [review]

File: src/app/kbv/controllers/question.js (hunk 6 above)
Comment (merlinc) [reply src/app/kbv/controllers/question.js:15]: https://github.com/HMPO/hmpo-form-wizard/blob/master/lib/controller/index.js#L117-L131

Configure gives you an empty method to override - so it's easy :)

File: src/app/kbv/controllers/load-question.test.js (hunk 5 above)
Comment (merlinc) [reply src/app/kbv/controllers/load-question.test.js]: :scissors:

File: src/app/details/controllers/done.js (hunk 3 above)
Comment (merlinc) [reply src/app/details/controllers/done.js:12]: Fixed :)

**Review by MattBidewell — APPROVED** [review]
LGTM! nice one

**Pull request merged**